types/defines/trace.d.ts

```diff
@@ -0,0 +1,190 @@
+declare namespace TailStream {
```

Comment:
Non-blocking: Some of these are already defined in `src/workerd/api/trace.h`, do they need an explicit definition here too? Asking non-rhetorically, it looks like we have both `EmailEventInfo` and `TraceItemEmailEventInfo` now in the generated types and the only difference is having a `readonly type: "email"` variable, maybe we need both.

Reply:
The `TraceItemEmailEventInfo` variation is one that is automatically generated from the types `api/trace.h` which we're not using here. As we iterate on the API it's quite possible these will diverge more from the legacy schema/types. I'd rather keep them separate so that they can evolve independently.